backend/command/src/items/events/item-edit-suggested.handler.ts

```diff
@@ -0,0 +1,272 @@
+    await this.insertSuggestion(editSuggestion);
+
+    this.logger.debug(
+      `${ItemEditSuggestedHandler.name} Insert took: ${
+        performance.now() - insertSuggestionStartTime
+      }ms`,
+    );
+
+    // Extract tags from item if they exist
+    let tags: SuggestionTag;
+    if (editSuggestion.updatedItemValues.tags) {
+      // Spread to copy by value, not as reference
+      tags = { ...editSuggestion.updatedItemValues.tags };
+      // Remove tags from initial => Values are not updated when updating item
+      delete editSuggestion.updatedItemValues.tags;
+    }
+    const updateItemStartTime = performance.now();
+
+    await this.updateItemWithoutTags(
+      editSuggestion.itemSlug,
+      editSuggestion.updatedItemValues,
+    );
+
+    this.logger.debug(
+      `${ItemEditSuggestedHandler.name} Item update (no tags) took: ${
+        performance.now() - updateItemStartTime
+      }ms`,
+    );
+    // No tags in suggestion => we are done here
+    if (!tags) {
+      this.logger.debug(
+        `Total duration of ${ItemEditSuggestedHandler.name} was ${
+          performance.now() - insertSuggestionStartTime
+        } ms`,
+      );
+      return;
+    }
+
+    const updateTagsStartTime = performance.now();
+    await this.updateTags(tags);
+    this.logger.debug(
+      `Updating tags by themselves took ${
+        performance.now() - updateTagsStartTime
+      } ms`,
+    );
+
+    const updateItemTagsStartTime = performance.now();
+    await this.updateItemTags(editSuggestion.itemSlug, tags);
+    this.logger.debug(
+      `Updating tags for item took ${
+        performance.now() - updateItemTagsStartTime
+      } ms`,
+    );
+
+    const updateItemsByTagStartTime = performance.now();
+
+    await this.positiveUpdateItemsByTags(editSuggestion.itemSlug, tags.push);
+    await this.negativeUpdateItemsByTags(editSuggestion.itemSlug, tags.pull);
+
+    this.logger.debug(
+      `Updating ItemsByTag took ${
+        performance.now() - updateItemsByTagStartTime
+      } ms`,
+    );
+
+    this.logger.debug(
+      `Total duration of ${ItemEditSuggestedHandler.name} was ${
+        performance.now() - insertSuggestionStartTime
+      } ms`,
+    );
+  }
+
+  // Db calls: 1 save()
+  async insertSuggestion(suggestion: EditSuggestion) {
+    try {
+      const insertedSuggestion = new this.suggestionModel(suggestion);
+      await insertedSuggestion.save();
+      this.logger.log(
+        `Edit suggestion inserted for item ${insertedSuggestion.itemSlug}`,
+      );
+    } catch (error) {
+      this.logger.error(
+        `Insert suggestion for item ${suggestion.itemSlug}: ${error}`,
+      );
+      throw new InternalServerErrorException("Couldn't insert suggestion");
+    }
+  }
+
+  // Db calls: 1 update()
+  async updateItemWithoutTags(slug: string, itemData: SuggestionItem) {
+    try {
+      await this.itemModel.findOneAndUpdate({ slug: slug }, itemData, {
+        new: true,
+      });
```

Comment:
```suggestion
      await this.itemModel.findOneAndUpdate({ slug: slug }, itemData);
```

As previously said, this line makes no sense if u don't store the value